### PR TITLE
Update Gradle to 7.0

### PR DESCRIPTION
**Pull request by thc202**
Update Gradle Wrapper and build to 7.0.
Update Spotless to 5.12.1.
Update add-on plugin to 0.5.0.
Update Node.JS Gradle plugin to 0.10.0-alpha.1.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -169,7 +169,7 @@ spotless {
     java {
         licenseHeaderFile("gradle/spotless/license.java")
 
-        googleJavaFormat().aosp()
+        googleJavaFormat("1.7").aosp()
```
Comment (kingthorin) [inline build.gradle.kts:172]: Just out of curiosity does this indicate a Java version or a version of the format? (Or something totally different)

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Build errors seem node/npm related.

File: build.gradle.kts (hunk 1 above)
Comment (thc202) [reply build.gradle.kts:172]: Version of the format, newer versions require Java 11.

**Comment by thc202** [comment]
The tasks need explicit ordering.

**Comment by thc202** [comment]
No, not that...

**Comment by thc202** [comment]
Should be good now, fixed the `PATH` for the npm tasks and made debug options optional for ZAP task.

**Comment by kingthorin** [comment]
Thanks!

**Review by kingthorin — APPROVED** [review]

**Review by psiinon — APPROVED** [review]

**Pull request merged**